src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencySubscriptionsHost.cs

```diff
@@ -101,21 +105,11 @@ internal sealed class DependencySubscriptionsHost : OnceInitializedOnceDisposedA
                 _dependenciesUpdateThrottleInterval,
                 commonServices.ThreadingService,
                 tasksService.UnloadCancellationToken);
-
-            ProjectFilePath = commonServices.Project.FullPath;
```

Comment:
This would be out of sync with rename, so good change.

Reply:
That's the motivation for the change. It resulted in a leak in `AggregateDependenciesSnapshotProvider` which keyed a dictionary on project path. There may have been other consequences I didn't see too.

On a local branch I remove the `ProjectFilePath` property from `DependencySubscriptionHost` completely, and source it directly from the snapshot, so we have a single source of truth.